cocos/tiles.py

```diff
@@ -325,24 +326,39 @@ def load_tmx(filename):
                 tileset = TileSet.from_atlas(name, firstgid, path, tile_width,
                                              tile_height, row_padding=spacing,
                                              column_padding=spacing)
-                # TODO consider adding the individual tiles to the resource?
-                tilesets.append(tileset)
-                resource.add_resource(name, tileset)
             elif c.tag == 'tile':
-                # add properties to tiles in the tileset
-                gid = tileset.firstgid + int(c.attrib['id'])
-                tile = tileset[gid]
-                props = c.find('properties')
-                if props is None:
-                    continue
-                for p in props.findall('property'):
-                    # store additional properties.
-                    name = p.attrib['name']
-                    value = p.attrib['value']
-                    # TODO consider more type conversions?
-                    if value.isdigit():
-                        value = int(value)
-                    tile.properties[name] = value
+                try:
+                    # add properties to tiles in the tileset
+                    gid = tileset.firstgid + int(c.attrib['id'])
+                    tile = tileset[gid]
+                    props = c.find('properties')
+                    if props is None:
+                        continue
+                    for p in props.findall('property'):
+                        # store additional properties.
+                        name = p.attrib['name']
+                        value = p.attrib['value']
+                        # TODO consider more type conversions?
+                        if value.isdigit():
+                            value = int(value)
+                        tile.properties[name] = value
+                except (KeyError, AttributeError):
```

Comment:
This is an exceptionally broad try block, those two errors could come from any number of lines above. What specific key and attribute access errors are being caught here?